### PR TITLE
[VM/SS] vm create --storage-sku updated. Can now specify individual disk storage account skus

### LINKED ISSUE #7319: vm/vmss create: --storage-sku should take a list for user to specify different values for os or data disks
This work gets particular important with the introduction of new sku as it can only be implied on data disks e.g. 
    1. `--storage-sku premium_lrs` (apply across to all disks)
    2.` --storage-sku os=premium_lrs 0=standard_lrs`  (os disk gets premium, the first data disk lun at 0 gets stanadard_lrs, and the rest goes with server defaults)

### PR DESCRIPTION
**Pull request by adewaleo**
Improved --storage-sku logic to allow specification of storage account types for individual disks.

Close #7319 

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by adewaleo** [comment]
TODO: 
- update --storage-sku help text with new usage.
- add tests

**Comment by tjprescott** [comment]
What does the before and after UX look like?

**Comment by adewaleo** [comment]
> What does the before and after UX look like?
@tjprescott ,
Previously:
```
   usage: --storage-sku SKU
```
Now:
```
 usage: --storage-sku SKU | --storage-sku all=SKU | --storage-sku "os=SKU 0=SKU 1=SKU..."
```

I need to update the help to show the new usage / format.

**Comment by yugangw-msft** [comment]
>  usage: --storage-sku SKU | --storage-sku all=SKU | --storage-sku "os=SKU 0=SKU 1=SKU..."

I don't think we need to support `all`, a valid sku value alone should be applied to all disks

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py [hunk 1]
```diff
@@ -250,8 +250,9 @@ def build_vm_resource(  # pylint: disable=too-many-locals
         disk_info=None, boot_diagnostics_storage_uri=None, ultra_ssd_enabled=None):
 
     os_caching = disk_info['os'].get('caching')
-    # TODO: split out the storage_sku for os disk(can't be ultra-ssd) and individual data disks
-    os_storage_sku = None if storage_sku == 'UltraSSD_LRS' else storage_sku
+    os_storage_sku = disk_info['os'].get('storageAccountType')
+    if os_storage_sku and os_storage_sku.lower() == 'UltraSSD_LRS'.lower():
+        os_storage_sku = None
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py]: I might get lost here, but to me we should either error or warn here, and the code should be moved to validators

Comment (yugangw-msft) [reply src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py]: Nit: `'UltraSSD_LRS'.lower()` can be `ultrassd_lrs`

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_vm_utils.py [hunk 2]
```diff
@@ -274,3 +286,41 @@ def get_storage_blob_uri(cli_ctx, storage):
             raise CLIError('{} does\'t exist.'.format(storage))
         storage_uri = storage_account.primary_endpoints.blob
     return storage_uri
+
+
+def get_sku_dict_from_string(sku_string):
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_vm_utils.py]: i would suggest you do like `update_write_accelerator_settings` does to stay consistent on processing those values.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py [hunk 3]
```diff
@@ -498,7 +497,11 @@ def _validate_vm_create_storage_account(cmd, namespace):
         storage_client = get_mgmt_service_client(cmd.cli_ctx, ResourceType.MGMT_STORAGE).storage_accounts
 
         # find storage account in target resource group that matches the VM's location
-        sku_tier = 'Premium' if 'Premium' in namespace.storage_sku else 'Standard'
+        sku_tier = 'Standard'
+        for sku in namespace.storage_sku:
+            if 'Premium' in sku:
+                sku_tier = 'Premium'
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py]: you can break here.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_vm_utils.py [hunk 4]
```diff
@@ -274,3 +287,33 @@ def get_storage_blob_uri(cli_ctx, storage):
...
+
+
+def update_disk_sku_info(info_dict, skus):
+    usage_msg = 'Usage:\n\t[--storage-sku SKU | --storage-sku "ID=SKU ID=SKU ID=SKU..."]\n' \
+                'where each ID is "os" or a 0-indexed lun.'
+
+    def _update(info, lun, value):
+        luns = info.keys()
+        if lun not in luns:
+            raise CLIError("Data disk with lun of {} doesn't exists".format(lun))
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_vm_utils.py]: typo: `doesn't exist`

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py [hunk 5]
```diff
@@ -338,7 +337,7 @@ def _build_storage_profile():
                     'createOption': 'fromImage',
                     'name': os_disk_name,
                     'caching': os_caching,
-                    'managedDisk': {'storageAccountType': storage_sku}
+                    'managedDisk': {'storageAccountType': os_storage_sku}
```
[the remark this answers is not in the document]
Comment (adewaleo) [reply src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py]: It is used in one more but I will just replace usages of the variable with the above.

**Pull request merged**